File: Lacewing-Unity/Assets/Lacewing/Code/Data/DroneCommand.cs
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace Lacewing.Data
{
	public readonly struct DroneCommand
	{
		[DataMember] public readonly PositionVector velocity;
		[DataMember] public readonly OrientationVector orientation;

		[JsonConstructor]
		public DroneCommand(PositionVector velocity, OrientationVector orientation)
		{
			this.velocity = velocity;
			this.orientation = orientation;
		}
	}
}

File: Lacewing-Python/lacewing/data/drone_command.py
import json

from lacewing.data.ovector import OVector
from lacewing.data.pvector import PVector


class DroneCommand:
    velocity: PVector
    orientation: OVector

    def __init__(self, velocity: PVector, orientation: OVector):
        self.velocity = velocity
        self.orientation = orientation

    def to_json(self):
        return json.dumps({'velocity': self.velocity.to_dict(), 'orientation': self.orientation.to_dict()})
